app/src/main/java/com/kms/cura/view/fragment/PatientHomeFragment.java

```diff
@@ -0,0 +1,211 @@
+package com.kms.cura.view.fragment;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Address;
+import android.location.Geocoder;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Spinner;
+
+import com.kms.cura.R;
+import com.kms.cura.controller.ErrorController;
+import com.kms.cura.model.Settings;
+import com.kms.cura.utils.GPSTracker;
+import com.kms.cura.view.UpdateSpinner;
+import com.kms.cura.view.adapter.CheckBoxAdapter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class PatientHomeFragment extends Fragment implements RadioGroup.OnCheckedChangeListener ,UpdateSpinner{
+
+    private EditText edtName, edtLocation;
+    private RadioGroup rdbtngroupLocation;
+    private RadioButton rdbtnCurrentLocation, rdbtnManualEnter;
+    private Spinner spnSpeciality;
+    private Button btnRegister;
+    private Context mContext;
+    private Activity activity;
+    private String currentLocation = null;
+    private boolean checked = true;
+    private ArrayList<String> speciality;
+    private UpdateSpinner updateSpinner;
+    private CheckBoxAdapter specialityAdapter;
+    private boolean[] specialitySelected = null; // it stores which specialities are chosen
+
+
+    public PatientHomeFragment() {
+        // Required empty public constructor
+    }
+
+    public static PatientHomeFragment newInstance(Context mContext, Activity activity) {
```

Comment:
Why make this a static method but not creating constructor with param (Context mContext, Activity activity) ?